File: readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>
#include <stdlib.h>

//Declare global functions
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int determine_grade(int letters, int words, int sentences);

int main(void)
{
    //Ask for text
    string text = get_string("Text: ");
    //Get letter count
    int letters = count_letters(text);
    //Get word count
    int words = count_words(text);
    //Get sentence count
    int sentences = count_sentences(text);
    //Get grade level
    int grade = determine_grade(letters, words, sentences);
    //Print grade level
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
    return 0;
}
//Calculate letters
int count_letters(string text)
{
    int letters = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}
//Calculate words
int count_words(string text)
{
    int words = 0;
    for (int i = 0, n = strlen(text); i <= n; i++)
    {
        if (isspace(text[i]) || text[i] == 0)
        {
            words++;
        }
    }
    return words;
}
//Calculate sentences
int count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == 46 || text[i] == 33 || text[i] == 63)
        {
            sentences++;
        }
    }
    return sentences;
}
//Calculate grade
int determine_grade(letters, words, sentences)
{
    float fletters = letters;
    float fwords = words;
    float fsentences = sentences;
    float L = ((fletters / fwords) * 100.0);
    float S = ((fsentences / fwords) * 100.0);
    float index = ((0.0588 * L) - (0.296 * S) - 15.8);
    int grade = round(index);
    return grade;
}

File: sentimental-readability/readability.py
import re


def main():
    text = get_text()
    letters = letter_count(text)
    words = (words_count(text)) + 1
    sentences = sentences_count(text)
    grade = grade_level(text, letters, words, sentences)
    print(grade)


# Ask user to input text
def get_text():
    text = input("Input text: ")
    return text


# Count number of alphabetical chars in the text
def letter_count(text):
    return len(re.findall('[a-zA-Z]', text))


# Count number of words in the text
def words_count(text):
    return len(re.findall(' ', text))


# Count numbe of sentences in the text
def sentences_count(text):
    return len(re.findall('[\?\.\!]', text))


# Calculate grade level from Coleman-Liau index
def grade_level(text, letters, words, sentences):
    L = letters / (words / 100)
    S = sentences / (words / 100)
    grade = round(0.0588 * L - 0.296 * S - 15.8)
    if grade < 1:
        return "Before Grade 1"
    elif grade > 16:
        return "Grade 16+"
    else:
        return f"Grade {grade}"


if __name__ == "__main__":
    main()